src/ILCompiler.Compiler/src/Compiler/SingleFileCompilationModuleGroup.cs

```diff
@@ -27,7 +27,11 @@ public override void AddCompilationRoots(IRootingServiceProvider rootProvider)
         {
             base.AddCompilationRoots(rootProvider);
 
-            AddCompilationRootsForExports((EcmaModule)_typeSystemContext.SystemModule, rootProvider);
+            foreach (var inputFile in _typeSystemContext.ReferenceFilePaths)
+            {
```

Comment:
@jkotas It feels like the distinction between what's an input file and what's a reference is rather blurry for single file compilation. If I e.g. use a NuGet package with a library that has some `NativeCallable` methods, should those be exposed? Or only when the `NativeCallable` methods are in my entrypoint assembly? What's the "entrypoint assembly" if I'm compiling a library?

Maybe we should add a method on `ModuleDesc` that lets us enumerate all the NativeCallable and RuntimeExport methods without loading all the types and methods (because scanning all the junk is indeed rather expensive) and go with this?